### PR TITLE
feat: MetricKit TestVersion 1

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

This PR adds a first iteration of sending MetricKitData of `MXCrashDiagnostic`, `MXCPUExceptionDiagnostic`, and `MXDiskWriteExceptionDiagnostic` to Sentry. This PRs aim is to have the code running in the iOS-Swift test app via TestFlight to get some real MetricKit data of `MXCPUExceptionDiagnostic`, and `MXDiskWriteExceptionDiagnostic`.

`MXCrashDiagnostic` exists solely for validating the stacktrace symbolication.

I purposely didn't add this to the changelog yet, cause it's even too early for an official experimental phase. 

#skip-changelog

## :bulb: Motivation and Context

First step for https://github.com/getsentry/sentry-cocoa/issues/1661

## :green_heart: How did you test it?

Unit tests and real device. Sadly I was not able to get sample payloads for `MXCPUExceptionDiagnostic` and `MXDiskWriteExceptionDiagnostic` yet.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Samples/iOS-Swift/iOS-Swift/AppDelegate.swift [hunk 1]
```diff
@@ -27,6 +30,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             options.profilesSampleRate = 1.0
             options.attachScreenshot = true
             options.attachViewHierarchy = true
+            if #available(iOS 14.0, *) {
+                options.enableMetricKit = true
+            }
```
Comment (brustolin) [inline Samples/iOS-Swift/iOS-Swift/AppDelegate.swift:35]: h: Some merge problem maybe. 
Enabling MetricKit for the second time here.

File: Sources/Sentry/SentryMetricKitIntegration.m [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+#import <SentryHexAddressFormatter.h>
+#import <SentryMechanism.h>
+#import <SentryMetricKitIntegration.h>
+#import <SentrySDK+Private.h>
+#import <SentryStacktrace.h>
+#import <SentryThread.h>
+
+#if SENTRY_HAS_METRIC_KIT
+
+#    if __has_include(<MetricKit/MetricKit.h>)
```
Comment (brustolin) [inline Sources/Sentry/SentryMetricKitIntegration.m:16]: m: do we need both tests?

File: Sources/Swift/MetricKit/SentryMXManager.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    
+    func didReceiveDiskWriteExceptionDiagnostic(_ diagnostic: MXDiskWriteExceptionDiagnostic, callStackTree: SentryMXCallStackTree, timeStampBegin: Date, timeStampEnd: Date)
+    
+    func didReceiveCpuExceptionDiagnostic(_ diagnostic: MXCPUExceptionDiagnostic, callStackTree: SentryMXCallStackTree, timeStampBegin: Date, timeStampEnd: Date)
+}
+
+@available(iOS 14.0, macOS 12.0, *)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+@objc public class SentryMXManager: NSObject, MXMetricManagerSubscriber {
```
Comment (brustolin) [inline Sources/Swift/MetricKit/SentryMXManager.swift:24]: l: You can user @objcMembers, and every function and property will already be objc compatible
```suggestion
@objcMembers public class SentryMXManager: NSObject, MXMetricManagerSubscriber {
```

**Review by brustolin — APPROVED** [review]
LGTM

A few comments to talk about, nothing big

File: Sources/Sentry/SentryMetricKitIntegration.m (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/SentryMetricKitIntegration.m]: Yep, I added a comment.

**Pull request merged**